test_rsync_to_rdisc.py

```diff
@@ -132,22 +144,20 @@ def mock_run_vcf_upload(class_mocker):
 
 class TestCheckRsync():
     def test_ok(self, set_up_test, mocker):
-        bgarray_log = Path(set_up_test['tmp_path']/"bgarray.log")
-        tmperror_empty = Path(set_up_test['tmp_path']/"empty.error")
         rsync_result = rsync_to_rdisc.check_rsync(
-            set_up_test['analysis1'], "Exomes", tmperror_empty, bgarray_log
+            set_up_test['analysis1'], set_up_test['analysis1_transfer_settings']
         )
         assert rsync_result == "ok"
-        assert not tmperror_empty.exists()
-        assert "No errors detected" in bgarray_log.read_text()
+        assert not Path(rsync_to_rdisc.settings.temp_error_path).exists()
+        assert "No errors detected" in Path(rsync_to_rdisc.settings.log_path).read_text()
 
     def test_temperror(self, mock_send_mail_transfer_state, set_up_test):
-        bgarray_log = Path(f"{set_up_test['tmp_path']}/bgarray.log")
+        rsync_to_rdisc.settings.temp_error_path = f"{rsync_to_rdisc.settings.wkdir}/temp_not_empty.error"
```

Comment:
I don't understand how this works...`set_up_test` is scoped on `session` level, which means it is executed only once per session and than reused if required more often. `set_up_test` changes the `rsync_to_rdisc.settings.temp_error_path` as well. 

Apparently, the tests succeed regardless. Is this by chance of order of execution, or can this be explained due to different scopes of the class-function....?

Reply:
Good catch, `@pytest.fixture(scope="class")` solves this.